src/Uno.Extensions.Navigation.UI/ServiceProviderExtensions.cs

```diff
@@ -11,43 +11,90 @@ public static void AttachServices(this Window window, IServiceProvider services)
 
 	internal static IServiceProvider RegisterWindow(this IServiceProvider services, Window window)
 	{
-		return services.AddInstance(window);
+		var sp = services.AddScopedInstance(window);
+		// Force a get on the dispatcher to make sure the instance is created on this
+		// thread, which should ensure the DispatcherQueue exists
```

Comment:
This assume the current implementation of the `Dispatcher` class (which do a `GetForCurrentThread()` when it can't resolve it from the `Window`). I would prefer to explicitly do something like `services.AddScopedInstance(new Dispatcher(window.Dispatcher));`. 

Also then, could we remove the `Window` from the `services`?